modules/RTC/TraceablePeerConnection.js

```diff
@@ -1224,25 +1222,31 @@ function extractSSRCMap(desc) {
             }
         }
         for (const ssrc of mLine.ssrcs) {
-            if (ssrc.attribute !== 'msid') {
+            if (!this._usesUnifiedPlan && ssrc.attribute !== 'msid') {
                 continue; // eslint-disable-line no-continue
             }
 
-            const msid = ssrc.value;
-            let ssrcInfo = ssrcMap.get(msid);
+            // Use the mediaType as key for the source map for unified plan clients since msids are not part of
+            // the standard and the unified plan SDPs do not have a proper msid attribute for the sources.
+            // Also the ssrcs for sources do not change for Unified plan clients since RTCRtpSender#replaceTrack is
+            // used for switching the tracks so it is safe to use the mediaType as the key for the TrackSSRCInfo map.
+            const key = this._usesUnifiedPlan ? mLine.type : ssrc.value;
+            const ssrcNumber = ssrc.id;
+            let ssrcInfo = ssrcMap.get(key);
 
             if (!ssrcInfo) {
                 ssrcInfo = {
                     ssrcs: [],
                     groups: [],
-                    msid
+                    msid: key
```

Comment:
is this `msid` field used later on (can it be removed from this structure) ? it's misleading to put not an msid under the name `msid`

Reply:
I checked if its being used anywhere before opening this PR but couldn't find anything using this field. Do you know why it was added in the first place if its not being used anywhere ? RN will be running in plan-b mode for a while so I didn't want to mess with that just in case.